### DISCUSSION TITLE
How will the new publishing play with class-based channel values?

### DISCUSSION THREAD

**Discussion by Midnighter** [nextflow-io/nextflow#5131] [Q&A]
I just listened to your podcast about the new nextflow features and I was wondering how that will play with class-based values? What I mean is perhaps a clarification is needed for the following in the docs:

> By default, all files emitted by the channel will be published into the specified directory. If a channel emits list values, any files in the list (including nested lists) will also be published. 

What if my output is a class where one of the attributes defines a file? Will that also be recognized and published?

**Comment by bentsherman** [comment]
Custom classes are not currently scanned during publishing, you'll have to transform custom types into tuples before publishing them. We will likely support this when we add type checking and record types.
